src/lib/tableProvider.ts

```diff
@@ -110,15 +123,15 @@ export function parquetDataFrame(from: AsyncBufferFrom, metadata: FileMetaData):
   return {
     header,
     numRows: Number(metadata.num_rows),
-    rows({ start, end, orderBy }: { start: number, end: number, orderBy?: OrderBy}) {
+    rows({ start, end, orderBy }) {
       if (orderBy?.length) {
         const numRows = end - start
         const wrapped = new Array(numRows).fill(null).map(() => resolvableRow(header))
 
         getSortIndex(orderBy).then(indices => {
           // Compute row groups to fetch
           for (const index of indices.slice(start, end)) {
-            const groupIndex = groupEnds.findIndex(end => index < end)
+            const groupIndex = groups.findIndex(({ groupEnd }) => index < groupEnd)
             fetchRowGroup(groupIndex)
```

Comment:
nit: rename to fetchVirtualRowGroup and virtualGroupIndex, to make it clear we're not fetching full row groups?